lib/tidewave/mcp/connection.ex

```diff
@@ -91,7 +91,16 @@ defmodule Tidewave.MCP.Connection do
   end
 
   def dispatch(pid, callback, args) do
-    GenServer.call(pid, {:dispatch, callback, args})
+    timeout =
+      case args do
+        %{"timeout" => timeout} when is_integer(timeout) ->
+          timeout
```

Comment:
This should actually be a bit greater, since when we pass a 30 second timeout to project_eval, we need to wait for it first.

```suggestion
          timeout + 1000
```